Source/Charts/Data/Implementations/Standard/ChartDataSet.swift

```diff
@@ -197,9 +198,11 @@ open class ChartDataSet: ChartBaseDataSet
     open override func entriesForXValue(_ xValue: Double) -> [ChartDataEntry]
     {
         let match: (ChartDataEntry) -> Bool = { $0.x == xValue }
```

Comment:
I believe this line should be:

    let match: (ChartDataEntry) -> Bool = { $0.x >= xValue }

Note the >= instead of ==.

Reply:
Can you explain why?

Reply:
Actually nevermind, sorry for bothering you! Please disregard this and other comments - I'll delete them.

My thinking was that `partitioningIndex` function requires that all values after the desired index match the expression, but after looking more carefully I see you've added the `partition` function before running `partitioningIndex` - I'm guessing your intention is to make it possible to search in non-ordered data. I'm a little worried about the performance here, but otherwise I think the implementation is correct, except one detail: comparing Doubles directly. This is probably bigger discussion about the library design, but basically due to Double representation in memory, we should never compare Doubles using the equality operator. This will break for example in a simple case: 0.3 != 0.1 + 0.2. But as I said, that's probably a separate discussion.